From: Christoph Hellwig <hch@lst.de>
Date: Tue, 12 Jul 2016 18:20:15 +0900
Subject: PCI: Switch msix_program_entries() to use pci_msix_desc_addr()
References: bnc#993388,FATE#321732
Patch-Mainline: v4.8-rc1
Git-commit: 12eb21de1f26233ad195ffe991697d61790f4193

Instead of relying on the msix_entry structure for the vector number, read
it from the msi_desc.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
Reviewed-by: Alexander Gordeev <agordeev@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/pci/msi.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/pci/msi.c b/drivers/pci/msi.c
index bd05ef2..70d800d 100644
--- a/drivers/pci/msi.c
+++ b/drivers/pci/msi.c
@@ -714,11 +714,9 @@ static void msix_program_entries(struct pci_dev *dev,
 	int i = 0;
 
 	for_each_pci_msi_entry(entry, dev) {
-		int offset = entries[i].entry * PCI_MSIX_ENTRY_SIZE +
-						PCI_MSIX_ENTRY_VECTOR_CTRL;
-
 		entries[i].vector = entry->irq;
-		entry->masked = readl(entry->mask_base + offset);
+		entry->masked = readl(pci_msix_desc_addr(entry) +
+				PCI_MSIX_ENTRY_VECTOR_CTRL);
 		msix_mask_irq(entry, 1);
 		i++;
 	}
-- 
1.8.5.6

